### DISCUSSION TITLE
Authentication: `REPORT_IS_LOGGED_IN` v.s. `LOG_IN`, & more

### DISCUSSION THREAD

**Discussion by hmaurer** [mattpocock/xstate-catalogue#27] [Machine Q&A]
Hello hello,

I was wondering what the thought process behind having separate events for `REPORT_IS_LOGGED_IN` and `LOG_IN` was (and similarly for `REPORT_IS_LOGGED_OUT` and `LOG_OUT`)? There's a difference in semantics but it seems subtle.

In addition:
- Should the machine be extended to respond to `REPORT_IS_LOGGED_OUT` in the `loggedIn` state, so that it could be sent when an authentication token expires?
- Should the machine be extended with a `loggingIn` state? (in which case the difference between `LOG_IN` and `REPORT_IS_LOGGED_IN` would become much starker).

**Comment by mattpocock** [comment]
https://www.loom.com/share/d683ebd657664886a92902b82c7dfdf9
